src/library-authoring/LibraryHome.tsx

```diff
@@ -27,14 +33,14 @@ const LibraryHome = ({ libraryId, tabList, handleTabChange } : LibraryHomeProps)
 
   const renderEmptyState = () => {
     if (componentCount === 0) {
-      return isFiltered ? <NoSearchResults /> : <NoComponents />;
+      return isFiltered ? <NoSearchResults /> : <NoComponents canEditLibrary={canEditLibrary} />;
     }
     return null;
   };
 
   return (
     <Stack gap={3}>
-      <LibraryRecentlyModified libraryId={libraryId} />
+      <LibraryRecentlyModified libraryId={libraryId} canEditLibrary={canEditLibrary} />
```

Comment:
You don't have to change all these components' props to pass `canEditLibrary` (or even `libraryId`) from one to another.

Instead, in the `EmptyStates` component, just use some code like this:

```tsx
const { libraryId } = useParams();
const { data: libraryData } = useContentLibrary(libraryId);
const canEditLibrary = libraryData?.canEditLibrary ?? false;
```

React Query will make sure that `useContentLibrary` is de-duplicated so it won't make any additional requests to the server.

Reply:
Updated: https://github.com/openedx/frontend-app-course-authoring/pull/1198/commits/91c745e907fe3ea92b4a996a1fe418c5883e3f18